src/site/xhp/APCCachedRenderable.php

```diff
@@ -9,36 +9,40 @@
  *
  */
 
+namespace HHVM\UserDocumentation\ui;
+
+use namespace Facebook\XHP;
+use namespace Facebook\XHP\Core as x;
 use type HHVM\UserDocumentation\LocalConfig;
 
-class APCCachedRenderable
-  implements \XHPUnsafeRenderable, \XHPAlwaysValidChild {
+final class APCCachedRenderable
+  implements XHP\UnsafeRenderable, XHP\AlwaysValidChild {
   private function __construct(private string $str) {
   }
 
-  public function toHTMLString(): string {
+  public async function toHTMLStringAsync(): Awaitable<string> {
     return $this->str;
   }
 
-  public static function get(string $key): ?\XHPRoot {
+  public static function get(string $key): ?x\node {
     $_success = null;
-    $str = apc_fetch(self::makeKey($key), inout $_success);
-    if (is_string($str)) {
+    $str = \apc_fetch(self::makeKey($key), inout $_success);
+    if (\is_string($str)) {
       $ret = <x:frag>{new APCCachedRenderable($str)}</x:frag>;
       return $ret;
     }
     return null;
   }
 
-  public static function store(string $key, \XHPRoot $content): void {
-    $str = $content->toString();
+  public static function store(string $key, x\node $content): void {
```

Comment:
Doesn't need to be in this diff, but this should be refactored to be an async function

Reply:
Yeah I started with that, but it's used from multiple functions and often in loops, so I left it for later.